From: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
Date: Sun, 2 Sep 2018 15:00:43 +0200
Subject: [PATCH] Remove BIO_callback_ctrl()

This removes BIO_callback_ctrl(), bio_info_cb and its two users.
This is the small version of #4220 ("Remove unused BIO bindings.").
OpenSSL changed this type because it was wrong and the (now) different
prototype fails to build.
There should be no users which rely on this function.

Link: https://github.com/pyca/cryptography/pull/4220
Signed-off-by: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
---
 src/_cffi_src/openssl/bio.py | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/src/_cffi_src/openssl/bio.py b/src/_cffi_src/openssl/bio.py
index a134d9b605b0..c4aa88c5b075 100644
--- a/src/_cffi_src/openssl/bio.py
+++ b/src/_cffi_src/openssl/bio.py
@@ -10,7 +10,6 @@ INCLUDES = """
 
 TYPES = """
 typedef struct bio_st BIO;
-typedef void bio_info_cb(BIO *, int, const char *, int, long, long);
 typedef ... bio_st;
 typedef ... BIO_METHOD;
 typedef ... BUF_MEM;
@@ -69,11 +68,6 @@ BIO *BIO_new_fp(FILE *, int);
 BIO *BIO_new_fd(int, int);
 BIO *BIO_new_socket(int, int);
 long BIO_ctrl(BIO *, int, long, void *);
-long BIO_callback_ctrl(
-    BIO *,
-    int,
-    void (*)(struct bio_st *, int, const char *, int, long, long)
-);
 long BIO_int_ctrl(BIO *, int, long, int);
 size_t BIO_ctrl_pending(BIO *);
 size_t BIO_ctrl_wpending(BIO *);
@@ -125,8 +119,6 @@ int BIO_set_close(BIO *,long);
 int BIO_get_close(BIO *);
 int BIO_pending(BIO *);
 int BIO_wpending(BIO *);
-int BIO_get_info_callback(BIO *, bio_info_cb **);
-int BIO_set_info_callback(BIO *, bio_info_cb *);
 long BIO_get_buffer_num_lines(BIO *);
 long BIO_set_read_buffer_size(BIO *, long);
 long BIO_set_write_buffer_size(BIO *, long);
-- 
2.11.0

